errdefs/notfound.go

```diff
@@ -0,0 +1,69 @@
+package errdefs
+
+import (
+	"errors"
+	"fmt"
+)
+
+// NotFound is an error interface which denotes whether the opration failed due
+// to a the resource not being found.
+type ErrNotFound interface {
+	NotFound() bool
```

Comment:
Note that this is a little different than github.com/cpuguy83/strongerrors which only defines such interfaces like so:

```go
type ErrNotFound interface {
  NotFound()
}
```

I haven't personally decided if I like it more, less or the same... probably about the same.

Reply:
The main advantage here is error implementations can short circuit the error traversal.

Reply:
@cpuguy83 Can you give an example of where short-circuiting / overriding error traversal would be valuable?

Reply:
So when we do something like:

```go
select {
case IsNotFound(err):
case IsInvalidInput(err):
case IsSomethingElse(err):
}
```

For each case (until it finds a match, of course), it will traverse the entire causal chain.

But if the err I pass in implements not found where it just returns `false`, it doeesn't need to traverse that chain.